lib/queuePopulator/LogReader.js

```diff
@@ -365,7 +372,18 @@ class LogReader {
                 batchState.publishedEntries[topic] = topicEntries;
                 return done();
             });
-        }, done);
+        }, err => {
+            // TODO: On error, produce error metrics
+            if (err) {
+                return done(err);
+            }
+            return async.each(this._extensions, (ext, cb) => {
```

Comment:
Since you don't wait for a `publishMetrics` callback, the calls are synchronous here so you should just use a simple `for` loop, or `this._extensions.forEach(...)`.

Reply:
With the above comment  (https://github.com/scality/backbeat/pull/125#discussion_r148858655), a callback has been added to the `publishMetrics` method 👍